WordPress/UITestsFoundation/TestObserver.swift

```diff
@@ -0,0 +1,63 @@
+import XCTest
+
+class TestObserver: NSObject, XCTestObservation {
+    override init() {
+            super.init()
+            XCTestObservationCenter.shared.addTestObserver(self)
+        }
+
+    func testBundleWillStart(_ testBundle: Bundle) {
+        // Code added here will run only once before all the tests
+        executeWithRetries(disableAutoFillPasswords)
+    }
+
+    func executeWithRetries(_ operation: () -> Bool) {
+        var retryCount = 3
+
+        while !operation() && retryCount > 0 {
+            retryCount -= 1
+        }
+    }
+
+    func disableAutoFillPasswords() -> Bool {
+        let settings = XCUIApplication(bundleIdentifier: "com.apple.Preferences")
+        settings.terminate()
+
+        settings.activate()
+        let passwordsMenuItem = settings.staticTexts["Passwords"]
+        passwordsMenuItem.waitForIsHittable()
+        guard passwordsMenuItem.waitForIsHittable() else {
+            XCTFail("SetUp Failed: Passwords menu item was not hittable in Settings.")
+            return false
+        }
+        settings.staticTexts["Passwords"].tap()
```

Comment:
There are a few `guards` like this which I'd like to DRY it but I couldn't use `waitAndTap( _ element: XCUIElement, maxRetries: Int = 10, timeout: TimeInterval = 10)` from [Globals.swift](https://github.com/wordpress-mobile/WordPress-iOS/blob/27bf04400b0382a1ec383d4f551996b50be4ff10/WordPress/UITestsFoundation/Globals.swift#L34). It worked when I added `JetpackUITests` Target Membership to Globals.swift, but It broke a dozen of other things. 😅

Reply:
or maybe we can add it as a private func in the same class to call in this method, it won't be reusable elsewhere but at least it's reusable here. wdyt?

Reply:
Yeah, I started doing it and then I noticed I was almost duplicating `waitForIsHittable()`, struggling with the Boolean returns to control the retry flow, and all of that because of the `XCTFail(String)` messages that would only help debugging locally. Since the local logs are enough to debug a possible issue, I decided to make it simple and rely on `waitForIsHittable()`. 
Changed in [`798de02`](https://github.com/wordpress-mobile/WordPress-iOS/pull/21148/commits/798de02aed721e6a34602d00ef8363ef49aee88c).